src/mappings/trade_reversions.ts

```diff
@@ -0,0 +1,48 @@
+import { log, BigInt, store } from '@graphprotocol/graph-ts'
+import { TradeReversion as TradeReversionEvent } from '../../generated/BatchExchange/BatchExchange'
+import { toEventId, toPriceId, getBatchId } from '../utils'
+import { createSolution, getSolutionById } from './solution'
+import { updateOrderOnTradeReversion } from './orders'
+import { getBatchById } from './batch'
+import { getTradeById } from './trades'
+
+export function onTradeReversion(event: TradeReversionEvent): void {
+  let batchId = getBatchId(event).minus(BigInt.fromI32(1))
+  let batch = getBatchById(batchId)
+  let solution = getSolutionById(batch.solution)!
+
+  // Delete price associated with that trade
+  let priceId = toPriceId(event.params.sellToken, batchId)
```

Comment:
Aaaaah, I see the issue now, getting a price entity requires a big int token Of and all we have is the entity token ID, which right now happens to be a string presentation of the former, but that would make the code slightly more brittle.